ext/jsonnet/vm.c

```diff
@@ -255,6 +257,112 @@ vm_set_max_trace(VALUE self, VALUE val)
     return Qnil;
 }
 
+static VALUE
+vm_set_fmt_indent(VALUE self, VALUE val)
+{
+    struct jsonnet_vm_wrap *vm = rubyjsonnet_obj_to_vm(self);
+    jsonnet_fmt_indent(vm->vm, NUM2INT(val));
+    return val;
+}
+
+static VALUE
+vm_set_fmt_max_blank_lines(VALUE self, VALUE val)
+{
+    struct jsonnet_vm_wrap *vm = rubyjsonnet_obj_to_vm(self);
+    jsonnet_fmt_max_blank_lines(vm->vm, NUM2INT(val));
+    return val;
+}
+
+static VALUE
+vm_set_fmt_string(VALUE self, VALUE val)
+{
+    char *str;
+    struct jsonnet_vm_wrap *vm = rubyjsonnet_obj_to_vm(self);
+    str = StringValueCStr(val);
+    if (str[0] == '\0' || str[1] != '\0' || (str[0] != 'd' && str[0] != 's' && str[0] != 'l')) {
+      rb_raise(rb_eArgError, "fmt_string only accepts 'd', 's', or 'l'");
+    }
+    jsonnet_fmt_string(vm->vm, str[0]);
+    return val;
```

Comment:
I suspect if this is more straightforward and slightly faster.

```suggestion
    struct jsonnet_vm_wrap *vm = rubyjsonnet_obj_to_vm(self);
    const char *ptr;
    if (RSTRING_LEN(str) != 1) {
        rb_raise(rb_eArgError, "fmt_string is too long");
    }
    ptr = RSTRING_PTR(str);
    switch (*ptr) {
    case 'd':
    case 's':
    case 'l':
        jsonnet_fmt_string(vm->vm, str[0]);
        return val;
    default:
        rb_raise(rb_eArgError, "fmt_string only accepts 'd', 's', or 'l'");
    }
```

Reply:
I think `StringValue(str)` is required before RSTRING_LEN/RSTRING_PTR to ensure str is String. So I've committed this change https://github.com/yugui/ruby-jsonnet/pull/17/commits/9bab98fa474ad7f76ae99f3c42c5b585b53cd566

Reply:
good point. thank you.